test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs

```diff
@@ -0,0 +1,187 @@
+            var testVersions = new List<VersionInfo>() {
+                new VersionInfo(new NuGetVersion(0, 0, 1)),
+                new VersionInfo(new NuGetVersion(0, 0, 2))
+            };
+            _testViewModel = new PackageItemListViewModel()
+            {
+                Version = testVersion,
+                Versions = new Lazy<Task<IEnumerable<VersionInfo>>> (() => Task.FromResult<IEnumerable<VersionInfo>>(testVersions)),
+                InstalledVersion = testVersion,
+            };
+
+            var resourceProvider = new MockNuGetResourceProvider();
+            var metadataResource = new MockPackageMetadataResource(_testData.TestData);
+            var repository = new MockSourceRepository<PackageMetadataResource>(
+                new PackageSource("nuget.psm.test"),
+                new List<INuGetResourceProvider>() { resourceProvider },
+                metadataResource);
+            var repositories = new List<SourceRepository>() { repository };
+            _metadataProvider = new MultiSourcePackageMetadataProvider(
+                repositories,
+                optionalLocalRepository: null,
+                optionalGlobalLocalRepositories: null,
+                logger: new Mock<ILogger>().Object);
+
+#pragma warning disable VSSDK005 // Avoid instantiating JoinableTaskContext
+            _joinableTaskContext = new JoinableTaskContext(Thread.CurrentThread, SynchronizationContext.Current);
+#pragma warning restore VSSDK005 // Avoid instantiating JoinableTaskContext
+            NuGetUIThreadHelper.SetCustomJoinableTaskFactory(_joinableTaskContext.Factory);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _joinableTaskContext?.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(true);
+        }
+    }
+
+    public class V3PackageDetailControlModelTests : V3DetailControlModelTestBase
+    {
+        private readonly PackageDetailControlModel _testInstance;
+
+        public V3PackageDetailControlModelTests(V3PackageSearchMetadataFixture testData)
+            : base(testData)
+        {
+            var solMgr = new Mock<ISolutionManager>();
+            _testInstance = new PackageDetailControlModel(
+                solutionManager: solMgr.Object,
+                nugetProjects: new List<NuGetProject>());
+            _testInstance.SetCurrentPackage(
+                _testViewModel,
+                ItemFilter.All,
+                () => null).Wait();
+        }
+
+        [Fact]
+        public async void PackageIsVulnerable_Flagged()
```

Comment:
Can this be renamed to better indicate why it should e flagged as vulnerable? Is it the because V3PackageSearchMetadataFixture configured both packages to be vulnerable?

Reply:
Good call--I'll go for something like `ViewModelMarkedVulnerableWhenMetadataHasVulnerability_Flagged()`. Thanks!